### PR TITLE
Transcribe A-star to Processing.

### PR DESCRIPTION
**Pull request by FriskyDev**
Several dynamic arrays here, so I used the ArrayList and its methods.

Can re-write for pure arrays on request.

### CODE REVIEW

File: CodingChallenges/CC_051_astar/Processing/CC_051_astar/Spot.pde [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      wall = true;
+    }
+  }
+  
+  // Display me
+  void show() {
+    if (wall) {
+      fill(0);
+      noStroke();
+      ellipse((float)i * w + w / 2.0, (float)j * h + h / 2.0, w / 2.0, h / 2.0);
```
Comment (shiffman) [inline CodingChallenges/CC_051_astar/Processing/CC_051_astar/Spot.pde]: Is the casting here necessary?

File: CodingChallenges/CC_051_astar/Processing/CC_051_astar/CC_051_astar.pde [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+// The road taken
+List<Spot> path = new ArrayList<Spot>();
+
+void setup() {
+  size(400, 400);
+  println("A*");
+
+  // Grid cell size
+  w = width / (float)cols;
```
Comment (shiffman) [inline CodingChallenges/CC_051_astar/Processing/CC_051_astar/CC_051_astar.pde]:
```suggestion
  w = float(width) / cols;
```

Processing has a `float` method which is admittedly a little weird, but makes the code a bit easier on the eyes for beginners.

File: CodingChallenges/CC_051_astar/Processing/CC_051_astar/CC_051_astar.pde [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  h = height / (float)rows;
```
Comment (shiffman) [inline CodingChallenges/CC_051_astar/Processing/CC_051_astar/CC_051_astar.pde]:
```suggestion
  h = float(height) / rows;
```

File: CodingChallenges/CC_051_astar/Processing/CC_051_astar/Spot.pde (hunk 1 above)
Comment (FriskyDev) [reply CodingChallenges/CC_051_astar/Processing/CC_051_astar/Spot.pde]: Hmm, I was actually hunting a bug when I did it. I think not, but I'll double check.
The issue was that the h variable in the class was shadowing the global one. So I changed the name of the variable. Removing the casts.

**Pull request merged**